code/controllers/subsystem/overlays.dm

```diff
@@ -217,35 +306,6 @@ SUBSYSTEM_DEF(overlays)
 	else
 		overlays |= cached_other
 
-
-/**
- * Compile all the overlays for an atom from the cache lists.
- * |= on overlays is not actually guaranteed to not add same appearances but we're optimistically using it anyway.
- */
-/atom/proc/compile_overlays()
-	var/list/ad = add_overlays
-	var/list/rm = remove_overlays
-
-	if(LAZYLEN(rm))
-		overlays -= rm
-		rm.Cut()
-
-	if(LAZYLEN(ad))
-		overlays |= ad
-		ad.Cut()
-
-	for(var/I in alternate_appearances)
-		var/datum/atom_hud/alternate_appearance/AA = alternate_appearances[I]
-		if(AA.transfer_overlays)
-			AA.copy_overlays(src, TRUE)
-
-	atom_flags &= ~ATOM_OVERLAY_QUEUED
-
-/atom/movable/compile_overlays()
-	. = ..()
-	UPDATE_OO_IF_PRESENT
-
-/turf/compile_overlays()
-	. = ..()
-	if (above)
-		update_above()
+/atom
+	var/tmp/list/our_overlays	//! our local copy of (non-priority) overlays without byond magic. Use procs in SSoverlays to manipulate
+	var/tmp/list/priority_overlays	//! overlays that should remain on top and not normally removed when using cut_overlay functions, like c4.
```

Comment:
Won't this set this instance as the first def of atom

Reply:
Probably, didn't think to check. Can check tomorrow when I'm back at my desktop.

Reply:
Checked, it's not changing where go-to-definition takes you for `/atom` -- it's already screwed up.